### PR TITLE
sql: make sql.txn.repeatable_read_isolation.enabled public

### PR DESCRIPTION
**Pull request by nvb**
This commit makes the `sql.txn.repeatable_read_isolation.enabled` cluster setting public, as part of moving REPEATABLE READ isolation into a public preview.

Epic: None

Release note (sql change): The cluster setting
`sql.txn.repeatable_read_isolation.enabled` was added and defaults to false. When set to true, the following statements will configure transactions to run under REPEATABLE READ isolation, rather than being automatically interpreted as SERIALIZABLE.
- `BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ`
- `SET TRANSACTION ISOLATION LEVEL REPEATABLE READ`
- `SET default_transaction_isolation = 'repeatable read'`
- `SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL REPEATABLE READ`

This setting was added since REPEATABLE READ transactions are a preview feature, so usage of it is opt-in for v24.3. In a future CockroachDB major version, this setting will change to default to true.

### CODE REVIEW

**Review by dikshant — APPROVED** [review]
LGTM 🚀

File: docs/generated/settings/settings-for-tenants.txt [hunk 1]
```diff
@@ -382,6 +382,7 @@ sql.ttl.default_select_batch_size	integer	500	default amount of rows to select i
 sql.ttl.default_select_rate_limit	integer	0	default select rate limit (rows per second) per node for each TTL job. Use 0 to signify no rate limit.	application
 sql.ttl.job.enabled	boolean	true	whether the TTL job is enabled	application
 sql.txn.read_committed_isolation.enabled	boolean	true	set to true to allow transactions to use the READ COMMITTED isolation level if specified by BEGIN/SET commands	application
+sql.txn.repeatable_read_isolation.enabled (alias: sql.txn.snapshot_isolation.enabled)	boolean	false	set to true to allow transactions to use the REPEATABLE READ isolation level. At the time of writing, this setting is intended only for usage by CockroachDB developers.	application
```
[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply docs/generated/settings/settings-for-tenants.txt:385]: Which part? The "application"? That's not a typo, it's a separate `Class` column.

But reading this makes me realize that we should change the description to not say `At the time of writing, this setting is intended only for usage by CockroachDB developers`. Will switch.

[the remark this answers is not in the document]
Comment (dikshant) [reply docs/generated/settings/settings-for-tenants.txt:385]: Ah missed the class header!

[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply docs/generated/settings/settings-for-tenants.txt]: > Will switch.

Done.

**Comment by nvanbenschoten** [comment]
TFTR!

bors r=dikshant

**Pull request merged**